oada-core/write-handler/oada-srvc-write-handler/write_handler.js

```diff
@@ -46,235 +46,236 @@ responder.on("request", (req, ...rest) => {
-          body = null;
-          changeType = "delete";
+          let ppath = Array.from(path)
+          method = (id, obj) => resources.deletePartialResource(id, ppath, obj)
+          body = null
+          changeType = 'delete'
         } else {
           if (!req.resourceExists)
-            return { rev: undefined, orev: undefined, changeId: undefined };
-          console.log("deleting resource altogether");
+            return { rev: undefined, orev: undefined, changeId: undefined }
+          console.log('deleting resource altogether')
           return resources
             .deleteResource(id)
             .tap(() =>
-              info("deleteResource", Date.now() / 1000 - beforeDeletePartial),
-            );
+              info('deleteResource', Date.now() / 1000 - beforeDeletePartial)
+            )
         }
       }
 
-      var obj = {};
-      var ts = Date.now() / 1000;
+      var obj = {}
+      var ts = Date.now() / 1000
       // TODO: Sanitize keys?
 
-      trace(req.resource_id+': Checking if resource exists (req.resourceExists = ',req.resourceExists,')');
+      trace(
+        req.resource_id +
+          ': Checking if resource exists (req.resourceExists = ',
+        req.resourceExists,
+        ')'
+      )
       if (req.resourceExists === false) {
-        console.log(
-          "initializing resource",
-          req.resource_id,
-          req.path_leftover,
-        );
-        id = req.resource_id.replace(/^\//, "");
-        path = path.slice(2);
+        console.log('initializing resource', req.resource_id, req.path_leftover)
+        id = req.resource_id.replace(/^\//, '')
+        path = path.slice(2)
 
         // Initialize resource stuff
         obj = {
-          _type: req["contentType"],
+          _type: req['contentType'],
           _meta: {
-            _id: id + "/_meta",
-            _type: req["contentType"],
-            _owner: req["user_id"],
+            _id: id + '/_meta',
+            _type: req['contentType'],
+            _owner: req['user_id'],
             stats: {
-              createdBy: req["user_id"],
-              created: ts,
-            },
-          },
-        };
-        trace('Intializing resource with ', obj);
+              createdBy: req['user_id'],
+              created: ts
+            }
+          }
+        }
+        trace('Intializing resource with ', obj)
       }
 
       // Create object to recursively merge into the resource
       if (path.length > 0) {
-        let o = obj;
-        let endk = path.pop();
+        let o = obj
+        let endk = path.pop()
         path.forEach(k => {
           if (!(k in o)) {
             // TODO: Support arrays better?
-            o[k] = {};
+            o[k] = {}
           }
-          o = o[k];
-        });
-        o[endk] = body;
+          o = o[k]
+        })
+        o[endk] = body
       } else {
-        obj = Object.assign(obj, body);
+        obj = objectAssignDeep(obj, body)
```

Comment:
Why doesn't `Object.assign` work here? Seems like we are at the bottom of the path at this point.

Reply:
Is this the only change?

We *need* a common formatter ... this diffs are very hard to read. I'm perfectly happy with `prettier`'s default.

Reply:
Makes it look like I did more work